pkg/controller/controller.go

```diff
@@ -1351,9 +1349,68 @@ func (s *Service) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReque
 	}, nil
 }
 
+func pauseMetroSession(ctx context.Context, metroSessionID string, arr *array.PowerStoreArray) (paused bool, err error) {
+	metroSession, err := arr.Client.GetReplicationSessionByID(ctx, metroSessionID)
+	if err != nil {
+		return false, fmt.Errorf("could not get metro replication session %s", metroSessionID)
+	}
+
+	// confirm the session is in a state we can pause from.
+	if metroSession.State != gopowerstore.RsStateOk &&
+		metroSession.State != gopowerstore.RsStateSynchronizing &&
+		metroSession.State != gopowerstore.RsStatePaused &&
+		metroSession.State != gopowerstore.RsStateSystemPaused &&
+		metroSession.State != gopowerstore.RsStateFractured {
+		return false, fmt.Errorf("could not pause the metro replication session, %s, because the session is not in expected state to pause", metroSession.ID)
```

Comment:
What happens here? Does K8s retry the resize after a while and until the volume is expanded? What about the impact to say kubectl edit ...?

Reply:
As we noticed, K8s retries the expand operation until it receives success response.
We added some safety checks based on API doc and to cover idempotency case.

Reply:
Correct, if any call fails for whatever reason during the csi ControllerExpandVolume call, kubernetes will retry the operation.
This check is to make sure the metro session is in a state that can be paused. There are are a few scenarios in which it would not be allowed.

Not sure I'm following on impact related to kubectl edit